File: optimalFile.py
import json


def move(offset_point, initial_points):
    results = []
    for point in initial_points:
        if isinstance(point, str):
            parsed = json.loads(point)
            parsed['x'] += offset_point['x']
            parsed['y'] += offset_point['y']
            results.append(parsed)
        else:
            point['x'] += offset_point['x']
            point['y'] += offset_point['y']
            results.append(point)


polyline = [
  {'x': 0, 'y': 0},
  {'x': 10, 'y': 10},
  '{"x": 20, "y": 20}',
  {'x': 30, 'y': 30},
]

print(move(offset_point=[{'x': 10, 'y': -5}], initial_points=polyline))

File: optimalFile.js
function move(offsetPoint, initialPoints) {
    let results = [];
    for (let element of initialPoints) {
        if (typeof element === 'object') {
            element.x += offsetPoint.x;
            element.y += offsetPoint.y;
            results.push(element);
        } else {
            let parsed = JSON.parse(element);
            parsed.x += offsetPoint.x;
            parsed.y += offsetPoint.y;
            results.push(parsed);
        }
    }
    return results;
}

const polyline = [
    { x: 0, y: 0 },
    { x: 10, y: 10 },
    '{ "x": 20, "y": 20 }',
    { x: 30, y: 30 },
];

console.log(move({ x: 10, y: -5 }, polyline))
